### ISSUE TITLE
LSP-json cannot seem to find npm

### ISSUE THREAD

**Issue by dasayan05** [sublimelsp/LSP-json#95]
#### System Info:
1. Ubuntu 18.04
2. Sublime Build 4107
3. `node` and `npm` installed at user level (don't have root permission) and visible in `${PATH}`

Just after installation, `LSP-json` showed an error pop-up

![Screenshot from 2021-06-07 13-49-36](https://user-images.githubusercontent.com/18719603/121020104-f3405200-c797-11eb-8edf-8be07629dedf.png)

.. and it is not working. I tried reinstalling and restarting, nothing worked. The sublime's console log shows this:

```
reloading python 3.3 plugin PackageDev._logging
reloading python 3.3 plugin PackageDev.main
reloading python 3.3 plugin Terminus.main
plugins loaded
Failed to register plugin ""
Traceback (most recent call last):
  File "/user/HS229/ad01424/.config/sublime-text-3/Installed Packages/LSP.sublime-package/plugin/core/sessions.py", line 661, in _register_plugin_impl
    if client_configs.add_external_config(name, settings, base_file, notify_listener):
  File "/user/HS229/ad01424/.config/sublime-text-3/Installed Packages/LSP.sublime-package/plugin/core/settings.py", line 37, in add_external_config
    config = ClientConfig.from_sublime_settings(name, s, file)
  File "/user/HS229/ad01424/.config/sublime-text-3/Installed Packages/LSP.sublime-package/plugin/core/types.py", line 560, in from_sublime_settings
    base = sublime.decode_value(sublime.load_resource(file))
  File "/user/HS229/ad01424/sublime_text/Lib/python33/sublime.py", line 435, in load_resource
    raise IOError('resource "%s" not found' % name)
OSError: resource "Packages//.sublime-settings" not found

Package Control: Skipping automatic upgrade, last run at 2021-06-07 12:50:47, next run at 2021-06-07 13:50:47 or after
Unable to open /user/HS229/ad01424/.config/sublime-text-3/Packages/LSP/LSP.sublime-settings
ignored packages updated to: ["LSP-json", "Vintage"]
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Package Control.sublime-settings
ignored packages updated to: ["Vintage"]
reloading python 3.3 plugin LSP-json.plugin
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Unable to start subprocess for LSP-json
Traceback (most recent call last):
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/server_npm_resource.py", line 88, in install_or_update
    self._node_runtime.npm_install(self._server_dest)
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/node_runtime.py", line 125, in npm_install
    stdout, error = run_command_sync(args, cwd=package_dir)
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/helpers.py", line 20, in run_command_sync
    output = subprocess.check_output(args, cwd=cwd, shell=sublime.platform() == 'windows', stderr=subprocess.STDOUT)
  File "./python3.3/subprocess.py", line 576, in check_output
  File "./python3.3/subprocess.py", line 819, in __init__
  File "./python3.3/subprocess.py", line 1448, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: 'npm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/user/HS229/ad01424/.config/sublime-text-3/Installed Packages/LSP.sublime-package/plugin/core/windows.py", line 310, in start_async
    plugin_class.install_or_update()
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/_client_handler/abstract_plugin.py", line 108, in install_or_update
    server.install_or_update()
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/server_npm_resource.py", line 91, in install_or_update
    raise Exception(error)
Exception: [Errno 2] No such file or directory: 'npm'

Unable to start subprocess for LSP-json
Traceback (most recent call last):
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/server_npm_resource.py", line 88, in install_or_update
    self._node_runtime.npm_install(self._server_dest)
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/node_runtime.py", line 125, in npm_install
    stdout, error = run_command_sync(args, cwd=package_dir)
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/helpers.py", line 20, in run_command_sync
    output = subprocess.check_output(args, cwd=cwd, shell=sublime.platform() == 'windows', stderr=subprocess.STDOUT)
  File "./python3.3/subprocess.py", line 576, in check_output
  File "./python3.3/subprocess.py", line 819, in __init__
  File "./python3.3/subprocess.py", line 1448, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: 'npm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/user/HS229/ad01424/.config/sublime-text-3/Installed Packages/LSP.sublime-package/plugin/core/windows.py", line 310, in start_async
    plugin_class.install_or_update()
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/_client_handler/abstract_plugin.py", line 108, in install_or_update
    server.install_or_update()
  File "/user/HS229/ad01424/.config/sublime-text-3/Packages/lsp_utils/st3/lsp_utils/server_npm_resource.py", line 91, in install_or_update
    raise Exception(error)
Exception: [Errno 2] No such file or directory: 'npm'
```

Kindly help. I an new to sublime and its packages.

**Comment by rchl** [comment]
You need to have `npm` installed. Typically this is installed automatically together with `node` with it seems like not in your case for some reason.

I think in your case it would be `sudo apt install npm`.

On my side I guess I should check for `npm` and offer local node installation if it's missing.

**Comment by dasayan05** [comment]
@rchl As I mentioned above (in the system info section) that I already have `npm` installed BUT at user level.
I had to do it because my system (DO NOT have root access) came with only `node`. So I installed fresh both `node` and `npm` at user level and prioritized them in the `$PATH`.

**Comment by dasayan05** [comment]
@rchl My system (do not have root access) indeed do not have `npm`. So I installed afresh a `node`+`npm` locally and prioritized in `$PATH`. I thought `LSP-json` can pick it up from `$PATH` but it seems to be not the case. Is there a quick hack by which I can point to my local `npm` binary?

**Comment by rchl** [comment]
https://lsp.sublimetext.io/troubleshooting/#updating-the-path-used-by-lsp-servers has information on how to handle the `$PATH` issues.

**Comment by dasayan05** [comment]
@rchl I figured it out. It wasn't particularly an `LSP-json` issue.
It was more related to the fact that launching sublime from Ubuntu GUI launcher cannot capture the `$PATH` properly as [discussed here](https://forum.sublimetext.com/t/why-doesnt-sublime3-use-global-path-variable-and-how-can-i-force-it-to-use-it/39392). That's why `LSP-json` couldn't figure out the custom `npm` path. I customized my `.desktop` launcher file to include the `npm` path in `$PATH` .. and it works normally.
